### PR TITLE
[#228] Documentation: Testing

### LINKED ISSUE #228: Documentation: Testing

### PR DESCRIPTION
**Pull request by Albanovukelaj17**

### CODE REVIEW

File: documentation/Writerside/documentation.tree [hunk 1]
```diff
@@ -43,5 +43,7 @@
             <toc-element topic="Repositories.md"/>
             <toc-element topic="Entities.md"/>
         </toc-element>
-    </toc-element>
+        <toc-element topic="Testing.md"/>
```
Comment (WalkerLukas) [inline documentation/Writerside/documentation.tree:46]: Put at corrent point in structure

File: documentation/Writerside/topics/Testing.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# Testing
+
+## Testing Overview
+The project was tested extensively in three primary modules: `common`, `server`, and `client`.
```
Comment (WalkerLukas) [inline documentation/Writerside/topics/Testing.md:4]: All three modules (`common`, `client`, `server`) were extensively tested.

File: documentation/Writerside/topics/Testing.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# Testing
+
+## Testing Overview
+The project was tested extensively in three primary modules: `common`, `server`, and `client`. 
+While `common` and `server` modules utilized automated testing through **JUnit**, 
+the `client` module relied on manual testing due to **GWT** limitations.
```
Comment (WalkerLukas) [inline documentation/Writerside/topics/Testing.md:6]: Please add the stuff we discussed

File: documentation/Writerside/topics/Testing.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+---
+
+## Common Module
+The [](Common.md) module contains shared models, DTOs, and utilities used across the application,both from Server and Client.
```
Comment (WalkerLukas) [inline documentation/Writerside/topics/Testing.md:11]: Comma error

File: documentation/Writerside/topics/Testing.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+This module was rigorously tested using **JUnit** to ensure the correctness and reliability of the shared functionality.
+
+### Key Areas of Testing
+- **[](Models.md):**
+    - Verified data integrity and edge cases in classes such as `GridModel`,`GridService`, `LobbyModels`,`PatternModel`,`PatternService`and `Vector2D`.
+- **DTOs:**
+    - Ensured correct serialization and deserialization of data through tests like `TestPatternListDTO`.
+- **Services:**
+    - Validated game logic and command verification processes in services such as [](Turn-Logic-Service.md) and `CommandValidator`.
```
Comment (WalkerLukas) [inline documentation/Writerside/topics/Testing.md:20]: Link to validator

File: documentation/Writerside/topics/Testing.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+- **Utilities:**
+    - Covered helper classes like `TestUtils` and `TurnResultNoOrder` to ensure robust utility methods.
```
Comment (WalkerLukas) [inline documentation/Writerside/topics/Testing.md:22]: TestUtils are used for testing. Not tested themself

File: documentation/Writerside/topics/Testing.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+### Testing Approach
+1. **Unit Tests:**
+    - Each method was tested to confirm expected outcomes.
```
Comment (WalkerLukas) [inline documentation/Writerside/topics/Testing.md:26]: Each method is an exaggeration. Dont upsell

File: documentation/Writerside/topics/Testing.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    - Tested for null inputs and extreme values to ensure stability and robustness.
+
+---
+
+## Server Module
+The [](Server.md) module is responsible for backend logic, WebSocket communication, and controller functionality. Testing in this module was automated using **JUnit**.
+
+### Areas Covered
+- **WebSocket Services:**
+    - Verified connection stability and proper handling of message payloads.
```
Comment (WalkerLukas) [inline documentation/Writerside/topics/Testing.md:37]: Really?

File: documentation/Writerside/topics/Testing.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+- **Controllers:**
+    - Ensured REST endpoints in classes like `HealthControllerTest` processed requests and responses correctly.
```
Comment (WalkerLukas) [inline documentation/Writerside/topics/Testing.md:39]: Ask marc for these tests

File: documentation/Writerside/topics/Testing.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+- **Handlers:**
+    - Validated event management and packet handling with tests such as `LobbyPacketHandlerTest` and `TextWebSocketHandlerExtTest`.
+- **Utilities:**
+    - Tested supporting classes like `EventTest` and `PairTest` to ensure reliability in backend operations.
+
+---
+
+## Client Module
+The [](Client.md) module relied on **manual testing** due to the limitations of **GWT** , which restricted the ability to perform automated tests.
```
Comment (WalkerLukas) [inline documentation/Writerside/topics/Testing.md:48]: Add the stuff we talked about

File: documentation/Writerside/topics/Testing.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+### Manual Testing Process
+- **Functional Testing:**
+    - Ensured UI responsiveness and verified gameplay features, such as character interactions and grid mechanics.
+- **Scenario-Based Testing:**
+    - Simulated different game states ( beginning, mid-game, and endgame) to test UI consistency and state transitions.
+- **Cross-Device Validation:**
+    - Confirmed compatibility and responsiveness on various devices and screen sizes.
+
+### Key Focus Areas
+1. Health bar rendering and updates.
```
Comment (WalkerLukas) [inline documentation/Writerside/topics/Testing.md:59]: This should not be on 1.
Add correct input and stuff
Didnt we talk about this?

**Review by WalkerLukas — CHANGES_REQUESTED** [review]
See comments

File: documentation/Writerside/topics/Testing.md (hunk 5 above)
Comment (Albanovukelaj17) [reply documentation/Writerside/topics/Testing.md:20]: to do

File: documentation/Writerside/topics/Testing.md (hunk 11 above)
Comment (Albanovukelaj17) [reply documentation/Writerside/topics/Testing.md:59]: that's an old version

File: documentation/Writerside/topics/Testing.md (hunk 9 above)
Comment (Albanovukelaj17) [reply documentation/Writerside/topics/Testing.md:66]: added him as reviewer

**Review by casqan — CHANGES_REQUESTED** [review]

**Review by casqan — DISMISSED** [review]

**Review by WalkerLukas — APPROVED** [review]
My issues were resolved. Rest is up to the others
